src/odemis/acq/stream/_static.py

```diff
@@ -686,6 +690,40 @@ def __init__(self, name, image, *args, **kwargs):
             cwl = (max_bw + min_bw) / 2
             width = (max_bw - min_bw) / 12
 
+        # Is there time data?
+        if image.shape[1] > 1:
+            self.temporal_data = True
+            try:
+                # cached list of wavelength for each pixel pos
+                self._tl_px_values = spectrum.get_time_per_pixel(image)
+            except (ValueError, KeyError):
+                # useless polynomial => just show pixels values (ex: -50 -> +50 px)
+                # TODO: try to make them always int?
+                max_t = image.shape[1] // 2
+                min_t = (max_t - image.shape[1]) + 1
+                self._tl_px_values = range(min_t, max_t + 1)
+                assert(len(self._tl_px_values) == image.shape[1])
+                unit_t = "px"
+                cwl = (max_t + min_t) // 2
+                width_t = image.shape[1] // 12
+            else:
+                min_t, max_t = self._tl_px_values[0], self._tl_px_values[-1]
+                unit_t = "s"
+                ct = (max_t + min_t) / 2
+                width_t = (max_t - min_t) / 12
+
+            # Create temproal data VA's
+
+            self.selected_time = model.FloatContinuous(self._tl_px_values[0],
+                                                   range=(min_t, max_t),
+                                                   unit=unit_t,
+                                                   setter=self._setTime)
+            # Use a nearest-neighbor algorithm to allow parsing values set
+            self.time_kdtree = scipy.spatial.KDTree([[x] for x in self._tl_px_values])
```

Comment:
I think it does need some nearest neighbor algorithm handled by the setter (not caller) because I had many issues with overlays, sliders, and other displays trying to set invalid values (between the values in the list) and it was simpler to just have the setter handle selecting the right values.